flake.nix

```diff
@@ -19,89 +17,309 @@
+        ];
+
         pkgs = import nixpkgs {inherit system overlays;};
+        unstable = import nixos-unstable {inherit system overlays;};
 
         rust-toolchain = (pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml).override {
-          extensions = ["cargo" "clippy" "rustfmt" "rust-src" "rust-std"];
-          targets = ["wasm32-unknown-unknown" "wasm32-wasi"];
-        };
+          extensions = [
+            "cargo"
+            "clippy"
+            "llvm-tools-preview"
+            "rust-src"
+            "rust-std"
+            "rustfmt"
+          ];
+
+          targets = [
+            "aarch64-apple-darwin"
+            "x86_64-apple-darwin"
 
-        nightly-rustfmt = pkgs.rust-bin.nightly.latest.rustfmt;
+            "x86_64-unknown-linux-musl"
+            "aarch64-unknown-linux-musl"
+
+            "wasm32-unknown-unknown"
+            "wasm32-wasi"
+          ];
+        };
 
         format-pkgs = with pkgs; [
           nixpkgs-fmt
           alejandra
+          taplo
+        ];
+
+        darwin-installs = with pkgs.darwin.apple_sdk.frameworks; [
+          Security
+          CoreFoundation
+          Foundation
         ];
 
         cargo-installs = with pkgs; [
+          cargo-criterion
           cargo-deny
           cargo-expand
           cargo-nextest
           cargo-outdated
           cargo-sort
           cargo-udeps
           cargo-watch
-          binaryen
+          llvmPackages.bintools
+          twiggy
+          unstable.cargo-component
           wasm-bindgen-cli
+          wasm-tools
         ];
-      in rec
-      {
-        devShells.default = pkgs.mkShell {
-          name = "rs-ucan";
 
-          # blst requires --target=wasm32 support in Clang, which MacOS system clang doesn't provide
-          stdenv = pkgs.clangStdenv;
+        cargo = "${pkgs.cargo}/bin/cargo";
+        node = "${unstable.nodejs_20}/bin/node";
+        wasm-pack = "${pkgs.wasm-pack}/bin/wasm-pack";
+      in rec {
+        devShells.default = pkgs.devshell.mkShell {
+          name = "ucan";
+
+          imports = [./pre-commit.nix];
 
-          nativeBuildInputs = with pkgs;
+          packages = with pkgs;
             [
-              # The ordering of these two items is important. For nightly rustfmt to be used instead of
-              # the rustfmt provided by `rust-toolchain`, it must appear first in the list. This is
-              # because native build inputs are added to $PATH in the order they're listed here.
-              nightly-rustfmt
-              rust-toolchain
-              pre-commit
-              protobuf
               direnv
+              rust-toolchain
               self.packages.${system}.irust
-              nodejs
-              nodePackages.pnpm
+              (pkgs.hiPrio pkgs.rust-bin.nightly.latest.rustfmt)
+
+              chromedriver
+              protobuf
+              unstable.nodejs_20
+              unstable.nodePackages.pnpm
+              unstable.wasmtime
```

Comment:
hmm, not sure why wasmtime is here?

Reply:
For debugging Wasm directly on the command line. It doesn't have to be there if it's a problem

Reply:
Ah, ok. I guess so. No problem, I just was wondering.